View dish18

mmollerstrom opened the pull request:
PT Story: [paste pivotal tracker link here]

##Changes proposed in this pull request:
  We created the relationship between the menu table and dish table.
  We created two new dishes.
  We printed out the dishes on the menu page.

##What I have learned working on this feature: 
  How to associate tables accurately.
  Illiterate over database record.

Screenshots:
<img width="1094" alt="screen shot 2017-04-21 at 12 43 26 pm" src="https://cloud.githubusercontent.com/assets/26405758/25274418/3dbdd65c-2690-11e7-9b98-ae3a4b67d2bd.png">
 

Ready for review!

@AmberWilkie

junydania committed:
Created step files nd feature test to view dishes

junydania committed:
Associated the Menu and dish table

mmollerstrom committed:
added features that allows user to see dishes from the database, test success

junydania committed:
Updated view dish feature

junydania committed:
view dishes page

mmollerstrom committed:
Merge branch 'develop' of https://github.com/CraftAcademy/slow_food_sinatra_april2017 into viewDish18

mmollerstrom committed:
resolves conflict, tuned up view, test success

AmberWilkie commented on features/step_definitions/view_dishes_steps.rb:
```diff
@@ -0,0 +1,14 @@
+
+Given(/^there is a dish called "([^"]*)" on "([^"]*)"$/) do |dishName, menuName|
```
This is nitpicking, but in Ruby we write variables like `dish_name` -> snake case

junydania replied:
Change applied

AmberWilkie commented on features/step_definitions/view_dishes_steps.rb:
```diff
@@ -0,0 +1,14 @@
+
+Given(/^there is a dish called "([^"]*)" on "([^"]*)"$/) do |dishName, menuName|
+  menu = Menu.create(name: menuName)
+  menu.dishes.create(name: dishName, menu_id: menu.id)
```
Does this work with just `menu: menu`? Try it.

junydania replied:
yes it worked.

AmberWilkie commented on features/view_dishes.feature:
```diff
@@ -0,0 +1,14 @@
+  Given I visit the Menu page
+  Then I should be able to see "Pizza"
+  And I should be able to see "Carbonara"
+  Then show me the page
```
Remove `show me the page` steps when you submit final code.

junydania replied:
Done

AmberWilkie commented on lib/controller.rb:
```diff
@@ -83,15 +83,16 @@ class SlowFood < Sinatra::Base
 
+  get '/menu' do
+    @menu = Menu.first
+    @dishes = Dish.all
```
Instead of searching for all dishes, we only want to see the dishes that are related to that menu. Remember `@menu.dishes`? Try it.

junydania replied:
change applied

AmberWilkie replied:
I still see `Dish.all`...

AmberWilkie replied:
Oh I see, you made two routes.... So what I meant is that instead of using the `Dish.all` database call, you should use `@menu.dishes`. Then in the view, loop through `@menu.dishes` instead of `@dishes`.

junydania replied:
Creating the two routes was an error. Single route only exist and has also been refactored as recommended

AmberWilkie commented on lib/models/dishes.rb:
```diff
@@ -0,0 +1,12 @@
+#require './models/menu'
```
Delete comments.

AmberWilkie commented on lib/views/menu.erb:
```diff
@@ -1 +1,4 @@
-<%= @menu.name %>
+<%= @menu.name %> <br>
+<% @dishes.each do |dish| %> <br>
```
Code formatting - indent this line like regular Ruby code would be indented.

junydania replied:
Done

AmberWilkie commented on spec/dish_spec.rb:
```diff
@@ -0,0 +1,8 @@
+
+  it { is_expected.to have_property :id }
+  it { is_expected.to have_property :name }
+
```
You need an `it { is_expected.to belong_to :menu }`

AmberWilkie reviewed (changes requested):
Good work guys - a few comments.

junydania committed:
changed the variable name used for creating menu and dish
deleted the then show me page action
corected the get menu section in the controller file
applied proper indentation in the menu.erb file
created a new test to check if menu exist int the database

junydania committed:
Pulled viewDish18 from Max and applied changes suggested in pull request

AmberWilkie commented on features/step_definitions/view_dishes_steps.rb:
```diff
@@ -0,0 +1,14 @@
+
+Given(/^there is a dish called "([^"]*)" on "([^"]*)"$/) do |dish, menu|
+  menu = Menu.create(name: menu)
```
Ack, sorry guys but here `menu` and `menu` are two different things, so you should change the name of one. Ruby allows this but other languages would not. It's bad practice.

junydania replied:
Corrected

junydania committed:
refactored the get menu route in the controller file

AmberWilkie reviewed (approved):
Nice work, guys.
